File: LeetCode/0704. Binary Search/Solution.java
/**
 * 46 / 46 test cases passed.
 * Runtime: 0 ms
 * Memory Usage: 39.5 MB 
 */
class Solution {
    public int search(int[] nums, int target) {
        int lo = 0, hi = nums.length - 1;
        while (lo <= hi) {
            int mid = lo + (hi - lo) / 2;
            if (nums[mid] == target) {
                return mid;
            }
            if (nums[mid] > target) {
                hi = mid - 1;
            } else {
                lo = mid + 1;
            }
        }
        return -1;
    }
}

File: LeetCode/0704. Binary Search/solution.py
"""
46 / 46 test cases passed.
Runtime: 32 ms
Memory Usage: 15.8 MB
"""
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        lo, hi = 0, len(nums) - 1
        while lo <= hi:
            mid = lo + (hi - lo) // 2
            if nums[mid] == target:
                return mid
            if nums[mid] > target:
                hi = mid - 1
            else:
                lo = mid + 1
        return -1
    
